use-cases/NVIDIA-GPU-passthrough-and-Kata.md

```diff
@@ -0,0 +1,296 @@
+      Region 3: Memory at 383c00000000 (64-bit, prefetchable) [size=32M]
+```
+
+non-large BARs devices:  Tesla T4, P4 ...
+```
+$ lspci -s 04:00.0 -vv | grep Region
+      Region 0: Memory at c4000000 (32-bit, non-prefetchable) [size=16M]
+      Region 1: Memory at 38ffc0000000 (64-bit, prefetchable) [size=256M] #less than 4G
+      Region 3: Memory at 38fff0000000 (64-bit, prefetchable) [size=32M]
+```
+
+The following steps outline the workflow for using a Nvidia GPU with Kata.
+## Host BIOS Requirements
+
+For large BARs devices, MMIO mapping above 4G address space should be enabled in the PCI configuration of the BIOS.
+Some hardware vendors use different names in BIOS, such as:
+
+- Above 4G Decoding
+- Memory Mapped I/O above 4GB
+- Memory Hole for PCI MMIO
+
+
+
+## Host Kernel Requirements
+The following configurations need to be enabled on your host kernel:
+
+- CONFIG_VFIO_IOMMU_TYPE1
+- CONFIG_VFIO
+- CONFIG_VFIO_PCI
+- CONFIG_VFIO_MDEV
+- CONFIG_VFIO_MDEV_DEVICE
+
+Your host kernel needs to be booted with `intel_iommu=on` on the kernel command line.
+
+## Install and configure Kata Containers
+To use this feature, you need Kata version 1.11.0 or above ([related PR](https://github.com/kata-containers/runtime/pull/2461)).
+Follow the [Kata Containers setup instructions](https://github.com/kata-containers/documentation/blob/master/install/README.md) to install the latest version of Kata.
+
+For non-large BARs devices (for example, Nvidia Tesla T4):
+The following configuration in the Kata `configuration.toml` file as shown below can work:
+```
+machine_type = "pc"
+
+hotplug_vfio_on_root_bus = true
+```
+
+For large BARs devices (for example, Nvidia Tesla P100):
+The following configuration in the Kata `configuration.toml` file as shown below can work:
+
+hotplug with **shpchp**:
+```
+machine_type = "q35"
+
+hotplug_vfio_on_root_bus = false
+```
+
+hotplug with **pciehp** ([related PR](https://github.com/kata-containers/runtime/pull/2410)):
+```
+machine_type = "q35"
+
+hotplug_vfio_on_root_bus = true
+pcie_root_port = 1
+```
+
+## Build Kata Containers kernel with GPU support
+The default guest kernel installed with Kata Containers does not provide GPU support. To use a Nvidia GPU with Kata Containers, you need to build a kernel with the necessary GPU support.
+The following kernel config options need to be enabled:
+```
+# Support PCI/PCIe device hotplug (Required for large BARs device)
+CONFIG_HOTPLUG_PCI_PCIE=y
+CONFIG_HOTPLUG_PCI_SHPC=y
+
+# Enable the MMIO access method for PCIe devices (Required for large BARs device)
+CONFIG_PCI_MMCONFIG=y
+
+# Support for loading modules (Required for load Nvidia drivers)
+CONFIG_MODULES=y
+CONFIG_MODULE_UNLOAD=y
+```
+
+The following kernel config options need to be disabled:
+```
+# Disable Open Source Nvidia driver nouveau
+# It conflicts with Nvidia official driver
+DRM_NOUVEAU=n
+```
+> Actually, nouveau is not enabled by default
```

Comment:
expand this note such as:

> **Note:** `CONFIG_DRM_NOUVEAU` is normally disabled by default. It is worth checking it is not enabled in your kernel configuration to prevent any conflicts.

Reply:
updated.